src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+        /// acceleration in G
+        /// </summary>
+        public float GyroscopeConvertion
+        {
+            get
+            {
+                float val = 0;
+                switch (GyroscopeRange)
+                {
+                    case GyroscopeRange.Range0250Dps:
+                        val = 250.0f;
+                        break;
+                    case GyroscopeRange.Range0500Dps:
+                        val = 500.0f;
+                        break;
+                    case GyroscopeRange.Range1000Dps:
+                        val = 1000.0f;
+                        break;
+                    case GyroscopeRange.Range2000Dps:
+                        val = 2000.0f;
+                        break;
+                    default:
+                        break;
+                }
+
+                val /= Adc;
+                // the sample rate diver only apply for the non FS modes
+                if ((GyroscopeBandwidth != GyroscopeBandwidth.Bandwidth3600HzFS32) &&
+                    (GyroscopeBandwidth != GyroscopeBandwidth.Bandwidth8800HzFS32))
+                    return val / (1 + SampleRateDivider);
+                return val;
+            }
+        }
+
+        /// <summary>
+        /// Get the gyroscope in degrees per seconds
+        ///    +Z   +Y
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///         +X
+        /// </summary>
+        public Vector3 Gyroscope => GetRawGyroscope() * GyroscopeConvertion;
+
+        private Vector3 GetRawGyroscope()
+        {
+            Span<byte> rawData = stackalloc byte[6] { 0, 0, 0, 0, 0, 0 };
+            Vector3 gyro = new Vector3();
+            ReadByteArray(Register.GYRO_XOUT_H, rawData);
+            gyro.X = BinaryPrimitives.ReadInt16BigEndian(rawData);
+            gyro.Y = BinaryPrimitives.ReadInt16BigEndian(rawData.Slice(2));
+            gyro.Z = BinaryPrimitives.ReadInt16BigEndian(rawData.Slice(4));
+            return gyro;
+        }
+
+        #endregion
+
+        #region Temperature
+
+        /// <summary>
+        /// Get the temperature
+        /// </summary>
+        public Temperature Temperature
+        {
+            get
+            {
+                Span<byte> rawData = stackalloc byte[2] { 0, 0 };
+                ReadByteArray(Register.TEMP_OUT_H, rawData);
+                // formula from the documentation
+                return Temperature.FromCelsius((BinaryPrimitives.ReadInt16BigEndian(rawData) - 21) / 333.87 + 21);
+            }
+        }
+
+        #endregion
+
+        #region Modes, constructor, Dispose
+
+        /// <summary>
+        /// Setup the Wake On Motion. This mode generate a rising signal on pin INT
+        /// You can catch it with a normal GPIO and place an interruption on it if supprted
+        /// Reading the sensor won't give any value until it wakes up preriodically
```

Comment:
several typos in the summary `supprted`, `availablein`, `Wake On Motion`. I'm not sure I understand what the paremeters should be and you don't have them documented, so that doesn't make it esasy to guess how to call this method.

Reply:
Yes, I've corrected a lot of typos, not only here but in the rest of the comments as well

Reply:
And yes, missing comments added!